trackbasket_be/models/krogerservice.py

```diff
@@ -1,115 +1,117 @@
 from flask_restful import Resource, reqparse, request
 from .at_risk_user import AtRiskUser
-import requests
-import os
-import json
+import requests, os, json
 
-proxyDict = {
-              "http"  : os.environ.get('FIXIE_URL', ''),
-              "https" : os.environ.get('FIXIE_URL', '')
-            }
+proxyDict = { "http"  : os.environ.get('FIXIE_URL', ''), "https" : os.environ.get('FIXIE_URL', '') }
 
 class Krogerservice:
 
   @classmethod
-  def return_token(cls):
+  def closest_store(cls, zipcode):
+    response  = cls.__store_api_request(zipcode)
+    store = cls.__extract_kroger_store(response)
+    if store is None:
+      return { 'error': 'no store found for this zipcode' }
+    return  { 'location_id': store['locationId'], 'name': store['chain'],'address': store['address']['addressLine1'],
+            'city': store['address']['city'],'state': store['address']['state'], 'zipcode': store['address']['zipCode'],
+            'latitude': store['geolocation']['latitude'],'longitude': store['geolocation']['longitude'] }
+  
+  @classmethod
+  def item_search(cls, term, at_risk_user_id):
+    nearest_store_id = cls.__get_store_id(at_risk_user_id)
+    if term == '':
+      return 'error'
+    json_items = cls.__item_api_request(term, nearest_store_id)
+    if json_items == []:
+      return 'error'
+    items = cls.__extract_items(json_items)
+    return { 'data': { 'id': 'items', 'attributes':  items } }
+
+  @classmethod
+  def __return_token(cls):
```

Comment:
made `return_token` and other methods private ('__'). Setting the `access_token` as a `global` variable we can access it from other methods

Reply:
interesting that we can still access private methods outside this class using 'name mangling'